components/CompanyRegister.tsx

```diff
@@ -1,128 +1,145 @@
 import { CreateCompanyItem } from '@/db/entities/company';
 import useCompany from '@/hooks/useCompanyRegistration';
 import { isSuccessResponse } from '@/pages/api/companies';
+import { zodResolver } from '@hookform/resolvers/zod';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
 import { toast } from 'react-toastify';
 import { useAccount } from 'wagmi';
+import { z } from 'zod';
 import { useRecruitmentRegisterCompany } from '../src/generated';
 import { Badge } from './ui/badge';
 
+const schema = z.object({
+  companyName: z
+    .string()
+    .min(1, { message: 'Company Name is required' })
+    .max(255),
+  companyEmail: z.string().email({ message: 'Invalid email format' }),
+});
+
+interface FormData {
+  companyName: string;
+  companyEmail: string;
+  companySite: string;
+}
+
 const CompanyRegister = () => {
   const router = useRouter();
   const { address } = useAccount();
-  const { isValidating, isRegistered } = useCompany(address);
+  const { isRegistered } = useCompany(address);
+  const [isValidating] = useState(false);
   const { isLoading, isSuccess, writeAsync } = useRecruitmentRegisterCompany();
 
-  const [companyName, setCompanyName] = useState('');
-  const [companyEmail, setCompanyEmail] = useState('');
-  const [companySite, setCompanySite] = useState('');
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    getValues,
+  } = useForm<FormData>({ resolver: zodResolver(schema) });
 
-  const registerCompanySc = async () => {
-    if (isRegistered) {
+  const onSubmit = async (data: FormData) => {
+    if (!isRegistered) {
+      try {
+        await writeAsync();
+        registerCompanyDb(data);
```

Comment:
missed this line again.
```suggestion

```

it is handled in the useEffect

Reply:
i left it and removed the useEffect as what it's just doing is to call registerDb function when the state of the form changes. But this appears to be a duplicated action

Reply:
we want your PR to be focussed purely on form validation.

By adding registerCompanyDb here you have changed the pattern used to trigger a DB save after a SC write.  That change is not in scope of your PR...

Please revert to the useEffect pattern that was already in place